### PR TITLE
Turndown: Fixes #5025: Convert inline katex html to its equivalent inline katex markdown

### LINKED ISSUE #5025: Katex gets broken when edit in wysiwyg  mode
Joplin 1.8.5 (prod, linux)  Ubuntu 20 LTS
Sync Version: 2
Profile Version: 35

## Steps to reproduce

1.  write some Katex mixed with texts
2.  go wysiwyg  mode and edit anythin on the page even a text will do
3.  then you should see the text changed, Katex lines sperated from text lines by itself and this will be very bad if you have lots of Katex in the page all will be messed up
4. this wont happen in markdown mode everything is OK

Peek 2021-05-29 06-45

### PR DESCRIPTION
**Pull request by Subhra264**
Fixes: #5025 

Currently, the turndown adds `'\n\n'`  around the content of `joplin-source` element assuming it as a block element( even if the `joplin-editable` node is a `span` element). As a result, the inline katex expression inside `joplin-editable` span element, gets converted to Md, but on a new line even though it is an inline expression.

This PR first checks if the `joplin-editable` element is an inline element, and then surrounds the content with relevant character.

## Tests :

katex_test

### CODE REVIEW

File: packages/turndown/test/index.html [hunk 1]
```diff
@@ -846,6 +846,40 @@ <h2>This is a header.</h2>
...
+    </div>
+  </div>
+  <pre class="expected">Hello World :
+
+$$
+\sqrt{3x}
+$$</pre>
+</div>
+
```
Comment (laurent22) [inline packages/turndown/test/index.html:882]: We don't use these tests anymore, they are from the original source code that we've forked, so please remove these changes and move anything relevant to html_to_md directory

Comment (Subhra264) [reply packages/turndown/test/index.html:882]: Ohh ok, I will do that

**Comment by Subhra264** [comment]
Moved the tests from `index.html` to `app-cli/tests/html_to_md` folder

**Comment by laurent22** [comment]
Looks good now, thanks for the fix @Subhra264!

**Pull request merged**